### PR TITLE
docs: generate docs/cmd from code

### LINKED ISSUE #129: Simplify and automate the documentation process
At present, all CLI documents are generated manually which 

- Makes doc and command help notes inconsistent
- Docs are kind of embedded in the code which also needs to decouple

To avoid such problems,
- Need to define a way to make the documentation process simple and automate the doc. the generation

### PR DESCRIPTION
**Pull request by vdemeester**
# Changes

- This uses `cobra` features to generate markdown and manpage documentation.
- This allows us to add example and longer description in separate
  markdown file if needed (in order to not cluter the code)

Man pages can be tested with `man --manpath=./docs/man tkn-pipeline-start` :angel: 

Closes #129 

/cc @sthaha @hrishin @chmouel 

Signed-off-by: Vincent Demeester <vdemeest@redhat.com>

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/cli/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

File: docs/reference/cmd/pipeline_start.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# pipeline start additional content
```
Comment (vdemeester) [inline docs/reference/cmd/pipeline_start.md]: This is an example of file to add more information and example :angel:

File: docs/cmd/tkn_pipeline_start.md [hunk 2]
```diff
@@ -1,38 +1,55 @@
...
+
+```shell
+$ tkn pipeline start --resource source=samples-git \
+	--resource image=my-image \
+	--param foo=yay \
+	--param bar=10
+```
 ```
 
 ### Options
```
Comment (hrishin) [inline docs/cmd/tkn_pipeline_start.md]: Just wondering why these options are getting print in the right format.
image

Comment (vdemeester) [reply docs/cmd/tkn_pipeline_start.md]: Ah interesting, there is a bug :joy: I'll look into why it does this…

**Comment by chmouel** [comment]
It's lgtm but i let @hrishin confirm his happy with it

**Comment by bobcatfish** [comment]
These look great! Maybe would be good to have link from README? 😇

**Comment by vdemeester** [comment]
@bobcatfish ahah that is a very good point !

**Pull request merged**